tests/components/evohome/conftest.py

```diff
@@ -150,6 +161,33 @@ async def setup_evohome(
 
         assert isinstance(mock_client.call_args.kwargs["session"], ClientSession)
 
-        assert mock_client.account_info is not None
+        assert evo and evo.account_info is not None
 
+        mock_client.return_value = evo
```

Comment:
This is a little cheeky. Is it OK to leave it like this?

The only other option is to yield a tuple, `mock_client, evo`, rather than just `mock_client`, but that would require me to refactor other tests.

Reply:
I am not exactly sure what your concern is. As in, what I now see is that you create your own client as a mock, and patched that into the creation of a new client object. I think that is fine.